linux-tools/test_time/test_time.py

```diff
@@ -0,0 +1,186 @@
+#!/bin/python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+# Copyright: 2016 IBM
+# Author: Basheer K<basheer@linux.vnet.ibm.com>
+
+import re
+import logging
+import time
+import os
+from autotest.client import test, utils
+from distutils.spawn import find_executable
+from autotest.client.shared import error
+
+
+class test_time(test.test):
+    """
+    Autotest module for time utility.
+    time - time is a simple command,it gives resource usage.
+    The time utility runs the specified program/command with the given
+    arguments.When program/command finishes,time writes a message
+    to standard error giving timing statistics about the command run.
+    These statistics consist of
+        (i)   The elapsed real time between invocation and termination,
+        (ii)  The user CPU time , and
+        (iii) The system CPU time.
+
+    @author:Basheer Khadarsabgari <basheer@linux.vnet.ibm.com>
+    """
+    version = 1
+    nfail = 0
+    seconds = 2
+    time_exe = ''
+    output_file = ''
+
+    def install_check(self):
+        """
+        Install check for time
+        """
+        self.time_exe = find_executable('time')
+        if self.time_exe is None:
+            raise error.TestError('time binary not found')
+
+    def test_portability_option(self):
+        """
+        This method will verify the time portability(-p) option
+        Ex: "/usr/bin/time -p sleep 2"
+        """
+        cmd = "%s -p sleep %s" % (self.time_exe, self.seconds)
+        ret = utils.run(cmd)
+        if re.search(
+                "(real\s+%s.\d+)\nuser\s+\d+.\d+\nsys\s+\d.\d+" %
+                self.seconds,
+                ret.stderr):
+            logging.info(
+                "Verified '--portability' option successfully.")
+        else:
+            self.nfail += 1
+            raise error.TestError(
+                "Failed to verify --portability option of time utility")
+
+    def test_format_option(self, cmd, expected_status):
+        """
+        This method will verify the time --format(-f) option
+        Ex:/usr/bin/time -f "%x is Exit status of the cmd %C" <cmd>
+        """
+        cmd_to_execute = self.time_exe + \
+            " -f '%x is the exit status of the cmd %C' " + cmd
+        search_pattern = "%s is the exit status of the cmd %s" % (
+            expected_status, cmd)
+        ret = utils.run(cmd_to_execute, ignore_status=True)
+        if search_pattern not in ret.stderr and \
+           ret.exit_status == expected_status:
+            self.nfail += 1
```

Comment:
Please correct the condition above. You are marking it pass based on "ret.exit_status == expected_status"
which is wrong.